plan/plans/select_list.go

```diff
@@ -140,6 +145,55 @@ func (s *SelectList) CloneHiddenField(name string, tableFields []*field.ResultFi
 	return false
 }
 
+// CheckReferAmbiguous checks whether an identifier reference is ambiguous or not in select list.
+// e,g, "select c1 as a, c2 as a from t group by a" is ambiguous,
+// but "select c1 as a, c1 as a from t group by a" is not.
+// For MySQL "select c1 as a, c2 + 1 as a from t group by a" is not ambiguous too,
+// so we will only check identifier too.
+// If no ambiguous, -1 means expr refers none in select list, else an index in select list returns.
+func (s *SelectList) CheckReferAmbiguous(expr expression.Expression) (int, error) {
```

Comment:
No, the main purpose is to check ambiguous, getting index is just an incidental feature.